lib/Async/include/Async/async.h

```diff
@@ -26,7 +28,10 @@ struct async_promise_base : async_registry::AddToAsyncRegistry {
   async_promise_base(std::source_location loc)
       : async_registry::AddToAsyncRegistry{std::move(loc)} {}
 
-  std::suspend_never initial_suspend() noexcept { return {}; }
+  std::suspend_never initial_suspend() noexcept {
+    _callerExecContext = ExecContext::currentAsShared();
```

Comment:
I wonder if this `callerExecContext` is actually worth doing 🤔

Reply:
I'm not sure what you mean.